src/gui/owncloudgui.cpp

```diff
@@ -250,45 +265,53 @@ void ownCloudGui::slotComputeOverallSyncStatus()
         _tray->setIcon(Theme::instance()->syncStateIcon(SyncResult::Paused, true, contextMenuVisible()));
         _tray->setToolTip(tr("Account synchronization is disabled"));
+        setStatusText(tr("Synchronization is paused"));
         return;
     }
 
     // display the info of the least successful sync (eg. do not just display the result of the latest sync)
     QString trayMessage;
     FolderMan *folderMan = FolderMan::instance();
     Folder::Map map = folderMan->map();
-    SyncResult overallResult = FolderMan::accountStatus(map.values());
+    SyncResult::Status overallResult = FolderMan::accountStatus(map.values()).status();
 
     // create the tray blob message, check if we have an defined state
-    if (overallResult.status() != SyncResult::Undefined) {
-        if (map.count() > 0) {
+    if (overallResult != SyncResult::Undefined && map.count() > 0) {
 #ifdef Q_OS_WIN
-            // Windows has a 128-char tray tooltip length limit.
-            trayMessage = folderMan->statusToString(overallResult.status(), false);
+        // Windows has a 128-char tray tooltip length limit.
+        trayMessage = folderMan->statusToString(overallResult.status(), false);
 #else
-            QStringList allStatusStrings;
-            foreach (Folder *folder, map.values()) {
-                QString folderMessage = folderMan->statusToString(folder->syncResult().status(), folder->syncPaused());
-                allStatusStrings += tr("Folder %1: %2").arg(folder->shortGuiLocalPath(), folderMessage);
-            }
-            trayMessage = allStatusStrings.join(QLatin1String("\n"));
-#endif
-        } else {
-            trayMessage = tr("No sync folders configured.");
+        QStringList allStatusStrings;
+        foreach (Folder *folder, map.values()) {
+            QString folderMessage = folderMan->statusToString(folder->syncResult().status(), folder->syncPaused());
+            allStatusStrings += tr("Folder %1: %2").arg(folder->shortGuiLocalPath(), folderMessage);
         }
+        trayMessage = allStatusStrings.join(QLatin1String("\n"));
+#endif
 
-        QIcon statusIcon = Theme::instance()->syncStateIcon(overallResult.status(), true, contextMenuVisible());
+        QIcon statusIcon = Theme::instance()->syncStateIcon(overallResult, true, contextMenuVisible());
         _tray->setIcon(statusIcon);
         _tray->setToolTip(trayMessage);
+
+        if (overallResult == SyncResult::Success || overallResult == SyncResult::Problem) {
+            setStatusText(tr("Up to date"));
+        } else if (overallResult == SyncResult::Paused) {
+            setStatusText(tr("Synchronization is paused"));
+        } else {
+            setStatusText(tr("Error during synchronization"));
+        }
     } else {
-        // undefined because there are no folders.
-        QIcon icon = Theme::instance()->syncStateIcon(SyncResult::Problem, true, contextMenuVisible());
+        if (overallResult == SyncResult::Undefined)
```

Comment:
this new behaviour is on purpose right?

Reply:
Actually, the behavior is identical. There were two branches before that lead to `no folders` and now there's only one because I unified them.